lib/connection.js

```diff
@@ -168,15 +163,33 @@ Connection.prototype.receiveResponse = function(data) {
 /**
  * Close this connection
  */
-Connection.prototype.close = function() {
+Connection.prototype.close = function(error) {
 
-    if (this.stream) {
+    if (!this.connected) {
+        return;
+    }
+
+    if (this.stream && this.stream.writable) {
         this.stream.end();
     }
+
     this.connected = false;
-    this.disconnected = true;
     this.stream = null;
     this.emit('close');
+
+    var hasRequests = this.requestBuffer.length;
+    if (hasRequests || error) {
+        this.emit('error.server', error || 'Server unexpectedly closed network connection.');
```

Comment:
So you're thinking we don't even reply to the responder in such a scenario? What I have right now has been working for us with this latest server issue.

The problem with only emitting an error is that it makes it impossible to reply to the exact request that first initiated it. I mean, there's a reason that the general node design is to always return the error object first. Since we can't do that for backwards compatibility, I feel like responder should be getting triggered with an error state.

Imagine a situation where the user receives a request in something like express, fires off a schema get request & then waits for the callback to return data back to the user. But if we never trigger that callback, their express request will just timeout & they won't be able to do any special handling of the Schema error.

This is what I have in my `hapi` app. But when Schema is unresponsive & doesn't call a callback, the request just times out, which is slow & painful for the user. If Schema replies then I can properly retry or show them the appropriate error.

```
{
    method: 'GET',
    path: API_BASE + 'page/{slug}',
    handler: (req, reply) => API.getPage(req).then(response => reply(response), error => reply(error).code(error.statusCode)),
  }
```